doc/experiment.rst

```diff
@@ -1,327 +1,423 @@
-others' data inadvertently. To prevent this situation, the SmartSim Entity object supports key
-prefixing, which automatically prepends the name of the model to the keys by which it is accessed.
-With this enabled, key overlapping is no longer an issue and ensemble members can use the same code.
-
-Under the hood, calling ensemble.enable_key_prefixing() causes the SSKEYOUT environment variable to
-be set, which in turn causes all keys generated by an ensemble member to be prefixed with its model
-name. Similarly, if the model for the ensemble member has incoming entities (such as those set via
-model.register_incoming_entity() or ensemble.register_incoming_entity()), the SSKEYIN environment
-variable will be set and the keys associated with those inputs will be automatically prefixed. Note
-that entities must register themselves as this is not done by default.
-
-Finally, please note that while prefixing is enabled by default for tensors, datasets, and aggregated
-lists of datasets, a SmartRedis client must manually call Client.use_model_ensemble_prefix() to
-ensure that prefixes are used with models and scripts.
-
-We modify the example above to enable key prefixing as follows:
-
-.. code-block:: bash
-
-  import numpy as np
-  from smartsim import Experiment
-
-  exp = Experiment("Training-Run", launcher="slurm")
-
-  # setup ensemble parameter space
-  learning_rate = list(np.linspace(.01, .5))
-  train_params = {"LR": learning_rate}
-
-  # define resources for all ensemble members
-  sbatch = exp.create_batch_settings(nodes=4,
-                                    time="01:00:00",
-                                    account="12345-Cray",
-                                    queue="gpu")
-
-  # define how each member should run
-  srun = exp.create_run_settings(exe="python",
-                                exe_args="./train-model.py")
-  srun.set_nodes(1)
-  srun.set_tasks(24)
-
-  ensemble = exp.create_ensemble("Training-Ensemble",
-                                params=train_params,
-                                params_as_args=["LR"],
-                                batch_settings=sbatch,
-                                run_settings=srun,
-                                perm_strategy="random",
-                                n_models=4)
-
-  # Enable key prefixing -- note that this should be done
-  # before starting the experiment
-  ensemble.enable_key_prefixing()
-
-  exp.start(ensemble, summary=True)
-
-
-Further Information
--------------------
-
-For more informtion about Ensembles, please refer to the :ref:`Ensemble API documentation <ensemble_api>`.
\ No newline at end of file
+Create a Ensemble
+^^^^^^^^^^^^^^^^^
+An ensemble is created through the function: ``Experiment.create_ensemble()``. The function requires
+one of the subsequent sets of arguments upon initialization:
+
+Case 1 : ``RunSettings`` and `params` or `replicas`
```

Comment:
Can we describe what these cases by what they would be used for rather than what type of data is sent in? This doesn't give a new user much guidance on how to pick

Reply:
Rough draft (something like this?):

If a user would like to run the Ensemble as a batch job, pass in a BatchSettings object...